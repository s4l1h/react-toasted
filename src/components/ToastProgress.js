import React from "react";

const ToastProgress = props => {
  return (
    <div
      className="toast-progress"
      style={{ width: props.percent.toString() + "%" }}
    />
  );
};
export default ToastProgress;
