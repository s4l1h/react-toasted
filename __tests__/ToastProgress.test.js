// __tests__/hidden-message.js
// these imports are something you'd normally configure Jest to import for you
// automatically. Learn more in the setup docs: https://testing-library.com/docs/react-testing-library/setup#cleanup
// import "@testing-library/react/cleanup-after-each";
import "@testing-library/jest-dom/extend-expect";

// NOTE: jest-dom adds handy assertions to Jest and is recommended, but not required

import React from "react";
import { render } from "@testing-library/react";
import ToastProgress from "../src/components/ToastProgress";

describe("ToastProgress Component", () => {
  test("it shows empty toast progress bar", () => {
    const { container } = render(<ToastProgress />);
    expect(container).toMatchInlineSnapshot(`<div />`);
  });
  test("it shows toast progress bar", () => {
    const { container } = render(<ToastProgress percent={90} />);
    expect(container).toMatchSnapshot();
  });
});
