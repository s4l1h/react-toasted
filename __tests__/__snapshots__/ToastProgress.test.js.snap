// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToastProgress Component it shows toast progress bar 1`] = `
<div>
  <div
    class="toast-progress"
    style="width: 90%;"
  />
</div>
`;
